src/daemon/emitter.cc

```diff
@@ -453,9 +472,18 @@ void Emitter::DoRemoteExecute(const base::WorkerPool& pool, ResolveFn resolver,
         counter.ReportOnDestroy(false);
         LOG(WARNING) << "Failed to connect to " << end_point->Print() << ": "
                      << error;
-        // Put into |failed_tasks_| to prevent hanging around in case all
-        // remotes are unreachable at once.
-        failed_tasks_->Push(std::move(*task));
+        bool& shard_switched = std::get<CHANGED_SHARD>(*task);
+        if (conf->emitter().has_total_shards() &&
+            conf->emitter().shard_queue_limit() && !shard_switched) {
+          // Let other shard complete task on connection failure. Do it once.
```

Comment:
Describe this logic better in comments: at least about `shard_queue_limit()` indicates enabled strict sharding that prevents this task from completion in case the remote server has gone.